sql: introduce a new virtual database crdb_internal

madelynnblue opened the pull request:
This exposes the following virtual tables:

- `tables`: table descriptor details.
- `schema_changes`: schema changes currently ongoing.
- `leases`: table leases held on the node where the query inspecting
  the `leases` table is ran.

These tables are intended for debugging / troubleshooting purposes,
and to provide additional information not present in pg_catalog or
information_schema.

Closes #10317

Code from @knz but reopened here as a simpler PR than that one.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/13718)
<!-- Reviewable:end -->

knz commented:
I'm obviously ok with this since I wrote it :)
lgtm but perhaps you want a green light from andrei too.

---

Reviewed 13 of 13 files at r1.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13718#-:-Kdez5Ap383iSgJodiP8:b-pjp3k)*
<!-- Sent from Reviewable.io -->

andreimatei commented:
I don't think my review is needed (and I'd rather skip it at the moment), but I'm very excited about this virtual database and looking forward to adding tables to it!

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13718#-:-Kdfv98d6eJPxwL-yGNe:b-nfq1eb)*
<!-- Sent from Reviewable.io -->

jordanlewis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> mod a bikeshed about the naming of `MODIFICATION_TIME_L`

---

Reviewed 13 of 13 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/sql/crdb_internal.go, line 43 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13718#-Kdg11E6sZZx6cpVOsZv:-Kdg11E6sZZx6cpVOsZw:bx1rn5r) ([raw file](https://github.com/cockroachdb/cockroach/blob/4b0555c2155c4f53bd08a54f043640fdd17c4202/pkg/sql/crdb_internal.go#L43)):*
> ```Go
>   VERSION                  INT NOT NULL,
>   MODIFICATION_TIME        TIMESTAMP NOT NULL,
>   MODIFICATION_TIME_L      DECIMAL NOT NULL,
> ```

What does _L mean? I guess logical, but that's not too obvious - I suggest renaming this to be more clear. Maybe MOD_TIME and MOD_TIME_LOGICAL?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13718#-:-KdgNalCw3uh15pUTOF8:b2nni7l)*
<!-- Sent from Reviewable.io -->

knz committed:
sql: introduce a new virtual database crdb_internal

This exposes the following virtual tables:

- `tables`: table descriptor details.
- `schema_changes`: schema changes currently ongoing.
- `leases`: table leases held on the node where the query inspecting
  the `leases` table is ran.

These tables are intended for debugging / troubleshooting purposes,
and to provide additional information not present in pg_catalog or
information_schema.

Closes #10317

mjibson commented:
Review status: 11 of 13 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/sql/crdb_internal.go, line 43 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13718#-Kdg11E6sZZx6cpVOsZv:-KdghNrDKGkArSUwBfYi:b-ho5wpv) ([raw file](https://github.com/cockroachdb/cockroach/blob/4b0555c2155c4f53bd08a54f043640fdd17c4202/pkg/sql/crdb_internal.go#L43)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

What does _L mean? I guess logical, but that's not too obvious - I suggest renaming this to be more clear. Maybe MOD_TIME and MOD_TIME_LOGICAL?
</blockquote></details>

Agreed. Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13718)*
<!-- Sent from Reviewable.io -->
